src/api/worker/search/DbFacade.ts

```diff
@@ -86,11 +86,11 @@ export class DbFacade {
 							}
 						}
 
-						DBOpenRequest.onupgradeneeded = (event: IDBVersionChangeEvent) => {
+						DBOpenRequest.onupgradeneeded = async (event: IDBVersionChangeEvent) => {
```

Comment:
This callback is not async and I think we'll have problems if we try to assume it's  async.

I wonder if we can block it here `Atomics.wait()` somehow, probably not without another worker?